### PR TITLE
config: remove the config item 'enable-dynamic-config'

### LINKED ISSUE #16357: remove the config item 'enable-dynamic-config'
## Development Task
After discussing, we decide to remove dynamic configs in TiDB and we did this in [PR/16022](https://github.com/pingcap/tidb/pull/16022).
Since this feature has been removed, we don't need the config item `enable-dynamic-config` now.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #16357 

Problem Summary: After discussing, we decide to remove dynamic configs in TiDB and we did this in PR/16022. Since this feature has been removed, we don't need the config item enable-dynamic-config now.

### What is changed and how it works?
Remove `enable-dynamic-config`.

### CODE REVIEW

File: executor/set_test.go [hunk 1]
```diff
@@ -945,7 +944,6 @@ func (s *testSuite10) TestSetConflictConfigItems(c *C) {
 	tk.MustExec("set tidb_enable_slow_log=1")
 	tk.MustQuery("select @@tidb_enable_slow_log").Check(testkit.Rows("1"))
 
-	config.GetGlobalConfig().EnableDynamicConfig = true
 	tk.MustExec("set tidb_slow_log_threshold=222")
 	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1105 cannot update tidb_slow_log_threshold when enabling dynamic configs"))
```
Comment (SunRunAway) [inline executor/set_test.go:948]: Should this case be an existence?

Comment (qw4990) [reply executor/set_test.go]: fixed, PTAL.

**Pull request merged**